docs/chainlink-keepers/compatible-contracts.md

```diff
@@ -11,28 +11,57 @@ whatsnext:
 
 ## Overview
 
-This guide explains how to make smart contracts **Keeper-compatible**. You will learn about the `KeeperCompatibleInterface` and its functions with an example contract. The guide showcases the convenience that Keepers provide to developers. To take full advantage of the Keepers automation infrastructure, read all of the documentation to understand the features of Chainlink Keepers.
+This guide explains how to make smart contracts **Keeper-compatible** by walking you through an example. You will learn about the `KeeperCompatibleInterface` and its functions with an example contract. The guide showcases the convenience that Keepers provide to developers. To take full advantage of the Keepers automation infrastructure, read all of the documentation to understand the features of Chainlink Keepers.
 
 **Table of Contents**
-
++ [Example Contract](#example-contract)
 + [Functions](#functions)
   + [`checkUpkeep` function](#checkupkeep-function)
     + [`checkData`](#checkdata)
     + [`performData`](#performdata)
   + [`performUpkeep` function](#performupkeep-function)
     + [`performData`](#performdata-1)
-+ [Example Contract](#example-contract)
+
+
+
+
+## Example Contract
+
+Let's look at the core elements of a Keepers-compatible contract, be examining a simple counter contract that will increment once a specified amount of time has passed. The Keepers Network will perform `checkUpkeep` off-chain during every block to see if `updateInterval` seconds has passed since the last timestamp. Once the specified amount of time has elapsed checkUpkeep will return true and the Keepers Network will execute `performUpkeep` on-chain to increment the counter.
+
+
+```solidity
+{% include samples/Keepers/KeepersCounter.sol %}
+```
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol" >Open in Remix</a>
+    <a href="/docs/conceptual-overview/#what-is-remix" > What is Remix?</a>
+</div>
+
+
+The basic requirements are:
+1. Importing KeepersCompatible.sol. You can refer to the [Chainlink Github](https://github.com/smartcontractkit/chainlink) to find the latest version.
+2. Using the **`KeepersCompatibleInterface`** from the library to ensure your `checkUpkeep` and `performUpkeep`function definitions are what our Keeper Network expects. 
+3. A **`CheckUpkeep` function** that specifies the logic to be checked off-chain to see if `performUpkeep` should be executed. This is a simulation so it can use on-chain data.
+4. A **`performUpkeep` function** that will be executed on chain when `CheckUpkeep` returns `true`.
+
+
+To see more complex examples, please refer to our [utility contracts](../utility-contracts) page.
+
```

Comment:
IMHO, it would be beneficial if we could give an indication to the reader that this is almost everything that they need to do and now it's time to make use of their keeper's contract.

The following is a suggestion:

### Run

Now it's time to compile and deploy your contract.
- In remix head over to the compile tab, hit the compile button and make sure that your contract compiles without any errors.
- Now head over to the deploy tab and deploy your smartcontract in an `injected web3` environment. When deploying your contract, enter a small interval in the constructor e.g 10. This will be the interval when the `performUpkeep` function will be called.
- Once deployment is complete make sure to copy the address of the deployed contract, this will be useful in the next step when [registering your upkeep](../register-upkeep/)

Reply:
Updated